pkg/endpoint/endpoint.go

```diff
@@ -1321,6 +1321,38 @@ func (e *Endpoint) GetPod() *slim_corev1.Pod {
 	return e.pod.Load()
 }
 
+// CEPOwnerInterface contains the interface of an endpoint owner.
+type CEPOwnerInterface interface {
```

Comment:
What about using the k8s [Object interface](https://pkg.go.dev/k8s.io/apimachinery@v0.29.1/pkg/apis/meta/v1#Object) instead?

Reply:
> What about using the k8s [Object interface](https://pkg.go.dev/k8s.io/apimachinery@v0.29.1/pkg/apis/meta/v1#Object) instead?

The issue is that it has way more methods that it's required as well as it doesn't provide the `IsNil()` method.